src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtPostcodeServiceTest.java

```diff
@@ -0,0 +1,148 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith({SpringExtension.class, MockitoExtension.class})
+@ContextConfiguration(classes = AdminCourtPostcodeService.class)
+public class AdminCourtPostcodeServiceTest {
+    private static final String COURT_SLUG = "test-slug";
+    private static final String NOT_FOUND = "Not found: ";
+
+    private static final int TEST_COURT_ID = 1;
+    private static final String TEST_POSTCODE1 = "M1";
+    private static final String TEST_POSTCODE2 = "M2";
+    private static final String TEST_POSTCODE3 = "M3";
+    private static final String TEST_POSTCODE4 = "M47ER";
+    private static final String NEW_POSTCODE = "M5";
+
+    private static final List<String> POSTCODES = Arrays.asList(
+        TEST_POSTCODE1,
+        TEST_POSTCODE2,
+        TEST_POSTCODE3,
+        TEST_POSTCODE4
+    );
+    private static final List<String> POSTCODES_TO_BE_DELETED = Arrays.asList(
+        TEST_POSTCODE2,
+        TEST_POSTCODE3
+    );
+    private static final List<String> POSTCODES_TO_BE_ADDED = Collections.singletonList(NEW_POSTCODE);
+    private static final int POSTCODE_COUNT = POSTCODES.size();
+
+    @Autowired
+    private AdminCourtPostcodeService adminService;
+
+    @MockBean
+    private CourtRepository courtRepository;
+
+    @MockBean
+    private CourtPostcodeRepository courtPostcodeRepository;
+
+    @Mock
+    private Court court;
+
+    private final List<CourtPostcode> courtPostcodes = Arrays.asList(
+        new CourtPostcode(TEST_POSTCODE1, court),
+        new CourtPostcode(TEST_POSTCODE2, court),
+        new CourtPostcode(TEST_POSTCODE3, court),
+        new CourtPostcode(TEST_POSTCODE4, court)
+    );
+
+    @Test
+    void shouldReturnAllCourtPostcodes() {
+        when(court.getCourtPostcodes()).thenReturn(courtPostcodes);
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+
+        assertThat(adminService.getCourtPostcodesBySlug(COURT_SLUG))
+            .hasSize(POSTCODE_COUNT)
+            .containsExactlyElementsOf(POSTCODES);
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenRetrievingPostcodesForNonExistentCourt() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> adminService.getCourtPostcodesBySlug(COURT_SLUG))
+            .isInstanceOf(NotFoundException.class)
+            .hasMessage(NOT_FOUND + COURT_SLUG);
+
+        verifyNoInteractions(courtPostcodeRepository);
+    }
+
+    @Test
+    void shouldAddCourtPostcodes() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(court));
+        when(courtPostcodeRepository.save(any())).thenReturn(new CourtPostcode(NEW_POSTCODE, court));
```

Comment:
Probably one that can be ignored, but should we add two or more postcodes to check that it works with multiples? May not be a biggie...probably isnt lol

Reply:
I think this should be fine to test for one postcode here. In AdminCourtPostcodeControllerTest, there is a very similar test addPostcodesShouldReturnCreatedStatus() which is testing adding of 3 postcodes. These tests in the controller and service levels are all broadly similar so no point repeating them for every single cases. Let me know if you disagree